### PR TITLE
qla publish, bump, & apps

### PR DESCRIPTION
**Pull request by jesseditson**
Adds a few commands:

- apps:
  - list, select, view apps and app versions
  - create new apps
- publish:
  - push a local build to the console
- bump:
  - bump version numbers on a local repo

### CODE REVIEW

**Comment by jesseditson** [comment]
Will add tests early next week, but ready for a look.

**Comment by jesseditson** [comment]
https://youtu.be/IejJfRX-bKM

**Review by gaspar09 — APPROVED** [review]

File: packages/quip-cli/src/lib/util.ts [hunk 1]
```diff
@@ -1,9 +1,77 @@
...
+                            } else {
+                                res([child]);
+                            }
+                        });
+                    });
+                })
+            ).then((childLists) => {
+                let arr: string[] = [];
+                // ts doesn't like [].concat, I think it believes it's immutable.
+                resolve(arr.concat(...childLists));
```
[the remark this answers is not in the document]
Comment (jesseditson) [reply packages/quip-cli/src/lib/util.ts:66]: Old habits I guess! Will update

File: packages/quip-cli/src/commands/publish.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+): Promise<Manifest | null> => {
+    const form = new FormData();
+    const {root, bundle, missing} = await createBundle(
+        manifest,
+        manifestPath,
+        ignore
+    );
+    if (missing.size > 0) {
+        println(chalk`{red WARNING: the following files were defined in your manifest, but were not found.}
+{red This bundle may be incomplete, you should include these files ore remove them from your manifest.}`);
```
[the remark this answers is not in the document]
Comment (jesseditson) [reply packages/quip-cli/src/commands/publish.ts:84]: ⛏

File: packages/quip-cli/src/lib/util.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jesseditson) [reply packages/quip-cli/src/lib/util.ts:66]: Oh right, cause this is an array of arrays. The ... translates to arr.concat(list1, list2, list3), which is equivalent to `childLists.reduce((all, child) => [...all, ...child], [])` - that might be more readable and I'd bet perf is similar, so I'll move to that.

**Comment by jesseditson** [comment]
Gonna do tests on another PR to avoid drift

**Pull request merged**